requestmatch/matcher.go

```diff
@@ -140,8 +159,8 @@ func makeLeaf(d Definition) (*leafMatcher, error) {
 		method:         d.Method(),
 		hostRxs:        hostRxs,
 		pathRxs:        pathRxs,
-		headersExact:   d.Headers(),
-		headersRegexps: allHeaderRxs,
+		headersExact:   keyValToLower(d.Headers()),
+		headersRegexps: headerRegexpsToLower(allHeaderRxs),
```

Comment:
there was some microoptimization on my mind: the initialization step doesn't matter, but the lookup matters, and I just simply considered the ToLower should be faster than CanonicalHeaderKey. but! this is still a good point, and not only because i didn't measure, but because ToLower is utf8 operation while there is a chance that CanonicalHeaderKey is not. Regardless, I make a quick measure, and update this.